### PR TITLE
feat: replace wasabi with logging

### LINKED ISSUE #34: Convert wasabi msg and printer usage to logging

### PR DESCRIPTION
**Pull request by MartinBernstorff**
- [X] I have assigned ranges (e.g. `>=0.1, <0.2`) to all new dependencies (allows dependabot to keep dependency ranges wide for better compatability)

Fixes #34.

# Todo
- [ ] Add logging to the other files

## Notes for reviewers
Logging is a pretty difficult beast!

One important aspect of logging is namespacing. Log messages will only be displayed if an appropriate logger has been set up to catch log messages from the library. 

Loggers are namespaced based on the file they are called from, such as `timeseriesflattener.flattened_dataset`. 

In order for log messages from this logger to be displayed, a handler must be instantiated within the parent namespace, e.g. `timeseriesflattener` or the root logger. The root logger can be instantiated by setting its name to "None", which is what I did in `test_flattened_dataset_generation.py`.

One exception: Log messages with a severity of warning or higher will automatically be printed to the standard output if they are not captured by another logger.

By defualt, flattened_dataset also logs to the root logger. This can be disabled with the `log_to_stdout` keyword.

Looks like this: 
image

### CODE REVIEW

File: .github/workflows/tutorials.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name: Check that tutorial notebooks work
+
+on: [push]
+
+jobs:
```
Comment (HLasse) [inline .github/workflows/tutorials.yml:5]: remove file

File: src/timeseriesflattener/feature_spec_objects.py [hunk 2]
```diff
@@ -16,6 +17,8 @@
 
 msg = Printer(timestamp=True)
```
Comment (HLasse) [inline src/timeseriesflattener/feature_spec_objects.py:18]: is this still used anywhere?

File: src/timeseriesflattener/feature_spec_objects.py [hunk 3]
```diff
@@ -38,7 +45,9 @@ def in_dict_and_not_none(d: dict, key: str) -> bool:
 
 def resolve_values_df(data: dict[str, Any]):
     """Resolve the values_df attribute of a feature spec to a values
-    dataframe."""
+
+    dataframe.
```
Comment (HLasse) [inline src/timeseriesflattener/feature_spec_objects.py:49]:
```suggestion
    dataframe.
```

File: src/timeseriesflattener/feature_spec_objects.py [hunk 4]
```diff
@@ -74,7 +83,9 @@ class BaseModel(PydanticBaseModel):
 
     class Config:
         """A pydantic basemodel, which doesn't allow attributes that are not
-        defined in the class."""
+
+        defined in the class.
```
Comment (HLasse) [inline src/timeseriesflattener/feature_spec_objects.py:87]:
```suggestion
        defined in the class.
```

File: src/timeseriesflattener/feature_spec_objects.py [hunk 5]
```diff
@@ -127,6 +138,7 @@ def get_col_str(self) -> str:
 
     def __eq__(self, other):
         """Trying to run `spec in list_of_specs` works for all attributes
+
```
Comment (HLasse) [inline src/timeseriesflattener/feature_spec_objects.py:141]: what's up with all these newlines in docstrings?

File: src/timeseriesflattener/flattened_dataset.py [hunk 6]
```diff
@@ -81,6 +74,7 @@ def __init__(  # pylint: disable=too-many-arguments
         predictor_col_name_prefix: str = "pred",
         outcome_col_name_prefix: str = "outc",
         n_workers: int = 60,
+        log_to_stdout: bool = True,
```
Comment (HLasse) [inline src/timeseriesflattener/flattened_dataset.py:77]: move __init__ to be the first method in the class

File: src/timeseriesflattener/logger.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
+"""Exampel of how to setup a logger with sensible defaults."""
```
Comment (HLasse) [inline src/timeseriesflattener/logger.py:1]:
```suggestion
"""Example of how to setup a logger with sensible defaults."""
```

File: src/timeseriesflattener/logger.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+import logging
+from pathlib import Path
+
+import coloredlogs
+
+from timeseriesflattener.utils import PROJECT_ROOT
+
+
+def setup_logger(
```
Comment (HLasse) [inline src/timeseriesflattener/logger.py:11]: This is surprisingly painful..

**Review by HLasse — COMMENTED** [review]
LGTM. Is anything still printed using wasabi (that is not warn or higher)? I.e., are there any informative messages that we would like to print regardless of whether the user has defined a logger?

File: src/timeseriesflattener/logger.py (hunk 8 above)
Comment (MartinBernstorff) [reply src/timeseriesflattener/logger.py:11]: This is also quite elaborate - a pretty good, easy example is:

```
logger = logging.getLogger(name)
coloredlogs.install()
```

And then you've got pretty decent defaults :-)

**Pull request merged**